### PR TITLE
Announce the starting items and spells for wanderers (11725), and start wanderers with a spell if they have only one level 1 spell

### PR DESCRIPTION
**Pull request by NormalPerson7**
First commit:

Previously, players that did not look at their M screen would not be
aware of if they had spells in their spell library.

Therefore, print the player's inventory and spell library to the message
log at the start of the game. Also, take a note of the spell library to
include in the dump. This is done by expanding the note_wanderer_items
function and renaming it to note_wanderer_equipment (to be more
accurate).

This commit also adds support to print any spells the player begins with
memorised at the start, although this currently never happens.

Second commit:

If wanderers have more than one level 1 spell, they may not have enough
spell levels to memorise all of them, and so the player should get to
choose which spell to memorise.

If, however, wanderers have only 1 level 1 spell in their library, and
they have a spell level available, it makes sense for the player to have
that spell automatically memorised. Hence, in this case, memorise the
player's only level 1 spell.

Please feel free to give comments on either of these commits, or any comments on how the code could be improved.

### CODE REVIEW

File: crawl-ref/source/main.cc [hunk 1]
```diff
@@ -560,16 +560,52 @@ static void _show_commandline_options_help()
...
                              }, ", ", ", ", mem_fn(&item_def::defined));
-    take_note(Note(NOTE_MESSAGE, 0, 0, equip_str));
+
+    // Wanderers start with at most 1 spell memorised.
+    const string spell_str =
+        !you.spell_no ? "" :
+        "; and the following spell memorised: "
+        + comma_separated_fn(begin(you.spells), end(you.spells),
+                             [] (const spell_type spell) -> string
+                             {
```
Comment (NormalPerson7) [inline crawl-ref/source/main.cc]: Perhaps this `comma_separated_fn` can be simplified to just `you.spells[0]` if wanderers start with at most 1 level 1 spell memorised? Should there also be an assert somewhere that you.spell_no = 0 or 1?

**Comment by MarvinPA** [comment]
I haven't had time to look into it properly but it doesn't look like memorise_wanderer_spell is working as intended, I rolled up a few characters and they could still start with a single L1 spell without it being memorised.

**Comment by NormalPerson7** [comment]
I tested it and it seemed to work for me, but in any case I'll do a bit more testing to check I didn't accidentally break something. It won't work if the player doesn't have any spell levels, perhaps that's what you were experiencing?

File: crawl-ref/source/ng-setup.cc [hunk 2]
```diff
@@ -509,6 +509,9 @@ static void _setup_generic(const newgame_def& ng)
     // Must be after _give_basic_knowledge
     add_held_books_to_library();
 
+    if (you.char_class = JOB_WANDERER)
```
Comment (MarvinPA) [inline crawl-ref/source/ng-setup.cc]: This should be an "==" rather than an "=".

File: crawl-ref/source/ng-wanderer.cc [hunk 3]
```diff
@@ -747,3 +747,24 @@ void create_wanderer()
...
 }
+
+void memorise_wanderer_spell()
+{
+    // If the player got only one level 1 spell, memorise it. Otherwise, let the
+    // player choose which spell(s) to memorise and don't memorise any.
+    auto const available_spells = get_sorted_spell_list(true, true);
+    if (available_spells.size())
+    {
+        int num_level_one_spells;
```
Comment (MarvinPA) [inline crawl-ref/source/ng-wanderer.cc]: It looks like the problem I was having with spells not memorising is this being uninitialised, initialising it at 0 makes it work as expected for me.

File: crawl-ref/source/ng-setup.cc (hunk 2 above)
Comment (NormalPerson7) [reply crawl-ref/source/ng-setup.cc]: Oh whoops, always need a second pair of eyes :)

File: crawl-ref/source/ng-wanderer.cc (hunk 3 above)
Comment (NormalPerson7) [reply crawl-ref/source/ng-wanderer.cc]: Hmm that's weird, does this vary between compilers?

Comment (rawlins) [reply crawl-ref/source/ng-wanderer.cc]: yes, it definitely can; and specificaly clang often behaves differently than gcc w.r.t. uninitialized variables.

Comment (NormalPerson7) [reply crawl-ref/source/ng-wanderer.cc]: That probably explains why mine worked fine then, thanks.

**Comment by MarvinPA** [comment]
Merged, thanks! (with the minor fixes squashed in). I think there's probably room to improve the messaging - it's a bit of a wall of text when you also have a bunch of spells, but it looks like it's all working as intended and is definitely a good improvement.

**Pull request closed without merging**